website/core/RemarkPlugins.js

```diff
@@ -0,0 +1,126 @@
+const u = require('unist-builder');
+
+function parseParams(paramString) {
+  var params = {};
+
+  if (paramString) {
+    var pairs = paramString.split('&');
+    for (var i = 0; i < pairs.length; i++) {
+      var pair = pairs[i].split('=');
+      params[pair[0]] = pair[1];
+    }
+  }
+
+  if (!params.platform) {
+    params.platform = 'web';
+  }
+
+  return params;
+}
+
+function htmlForCodeBlock(code) {
+  return (
+    '<pre><code class="hljs css javascript">' +
+    hljs.highlight('javascript', code).value +
+    '</code></pre>'
+  );
+}
+
+/**
+ * Use the SnackPlayer by including a ```SnackPlayer``` block in markdown.
+ *
+ * Optionally, include url parameters directly after the block's language.
+ * Valid options are name, description, and platform.
+ *
+ * E.g.
+ * ```SnackPlayer platform=android&name=Hello%20world!
+ * import React from 'react';
+ * import { Text } from 'react-native';
+ *
+ * export default class App extends React.Component {
+ *   render() {
+ *     return <Text>Hello World!</Text>;
+ *   }
+ * }
+ * ```
+ */
+function SnackPlayer() {
+  return tree =>
+    new Promise(async (resolve, reject) => {
+      const nodesToChange = [];
+      // Parse all CodeBlocks
+      visit(tree, 'code', (node, parent) => {
+        //Add SnackPlayer CodeBlocks to processing queue
+        if (node.lang == 'SnackPlayer') {
+          nodesToChange.push({
+            node,
+            parent,
+          });
+        }
+      });
+      for (const {node, parent} of nodesToChange) {
+        try {
+          let params = parseParams(node.meta);
+
+          // Gather necessary Params
+          const name = params.name
+            ? decodeURIComponent(params.name)
+            : 'Example';
+          const description = params.description
+            ? decodeURIComponent(params.description)
+            : 'Example usage';
+          const sampleCode = node.value;
+          const encodedSampleCode = encodeURIComponent(sampleCode);
+          const platform = params.platform ? params.platform : 'ios';
+          const supportedPlatforms = params.supportedPlatforms
+            ? params.supportedPlatforms
+            : 'ios,android,web';
+
+          // Generate Node for SnackPlayer
+          const snackPlayerDiv = u('html', {
+            value: `<div class="snack-player">
+                          <div class="mobile-friendly-snack" style="display: none">
+                          ${htmlForCodeBlock(sampleCode)}</div>
+                          <div class="desktop-friendly-snack" style="margin-top: 15px; margin-bottom: 15px">
+                          <div
+                            data-snack-name="${name}"
+                            data-snack-description="${description}"
+                            data-snack-code="${encodedSampleCode}"
+                            data-snack-platform="${platform}"
+                            data-snack-supported-platforms=${supportedPlatforms}
```

Comment:
should we add quotes there?

Reply:
Yup, basically the plugin injects HTML in place of the SnackPlayer code blocks so
`data-snack-description="${description}"` will be `data-snack-description="Description of the snackplayer"` in the page

Reply:
I mean `data-snack-supported-platforms` specifically does not have quotes will the others have, is it normal?

Reply:
Added them in 574838e